packages/flutter/lib/src/painting/box_decoration.dart

```diff
@@ -79,7 +79,7 @@ class BoxDecoration extends Decoration {
     this.boxShadow,
     this.gradient,
     this.shape: BoxShape.rectangle,
-  });
+  }) : assert(shape != null);
```

Comment:
Is it worth explicitly documenting that `[shape] must not be null`?

Reply:
Fixed. Also improved other documentation while I was at it.